test/830_cni_plugin_test.sh

```diff
@@ -85,6 +113,24 @@ assert_raises "[ $C2IP != $C3IP ]"
 assert_raises "[ $BRIP != $C3IP ]"
 assert_raises "exec_on $HOST1 c1 $PING $C3IP"
 
+# get the value of hairpin on the veth for each of ch0, ch1, ch2
+CH0_PEER_ID=$(exec_on $HOST1 ch0 cat /sys/class/net/eth0/iflink)
+CH1_PEER_ID=$(exec_on $HOST1 ch1 cat /sys/class/net/eth0/iflink)
+CH2_PEER_ID=$(exec_on $HOST1 ch2 cat /sys/class/net/eth0/iflink)
+
+CH0_PEER=$(run_on $HOST1 ip link show | awk "/^${CH0_PEER_ID}:/"' {print $2}' | sed 's/@.*$//g' )
+CH1_PEER=$(run_on $HOST1 ip link show | awk "/^${CH1_PEER_ID}:/"' {print $2}' | sed 's/@.*$//g' )
+CH2_PEER=$(run_on $HOST1 ip link show | awk "/^${CH2_PEER_ID}:/"' {print $2}' | sed 's/@.*$//g' )
+
+CH0_HAIRPIN=$(run_on $HOST1 cat /sys/devices/virtual/net/weave/brif/$CH0_PEER/hairpin_mode)
+CH1_HAIRPIN=$(run_on $HOST1 cat /sys/devices/virtual/net/weave/brif/$CH1_PEER/hairpin_mode)
+CH2_HAIRPIN=$(run_on $HOST1 cat /sys/devices/virtual/net/weave/brif/$CH2_PEER/hairpin_mode)
+
+# now we can compare what the modes should be
+assert_raises "[ $CH0_HAIRPIN == 1 ]"
+assert_raises "[ $CH1_HAIRPIN == 0 ]"
+assert_raises "[ $CH2_HAIRPIN == 1 ]"
```

Comment:
Tests cleaned up to use your "underhanded knowledge". :-)

I just added the comments about how you would go about it the long (and less fragile) way should you ever need to. If it ever breaks, save someone the work.